wicket/src/wicketd.rs

```diff
@@ -64,18 +79,120 @@ impl WicketdManager {
     /// * Receive responses / errors
     /// * Translate any responses/errors into [`Event`]s
     ///   that can be utilized by the UI.
-    pub async fn run(self) {
-        let mut inventory_rx =
-            poll_inventory(&self.log, self.inventory_client).await;
-
-        // TODO: Eventually there will be a tokio::select! here that also
-        // allows issuing updates.
-        while let Some(event) = inventory_rx.recv().await {
-            // XXX: Should we log an error and exit here? This means the wizard
-            // died and the process is exiting.
-            let _ = self.wizard_tx.send(Event::Inventory(event));
+    pub async fn run(mut self) {
+        self.poll_inventory().await;
+        self.poll_update_log().await;
+        self.poll_artifacts().await;
+
+        loop {
+            tokio::select! {
+                Some(request) = self.rx.recv() => {
+                    slog::info!(self.log, "Got wicketd req: {:?}", request);
+                    let Request::StartUpdate(component_id) = request;
+                    self.start_update(component_id).await;
+                }
+                else => {
+                    slog::info!(self.log, "Request receiver closed. Process must be exiting.");
+                    break;
+                }
+            }
```

Comment:
Can this just be the simpler `while let Some(event) = self.rx.recv.await()` that it used to be?

Reply:
Ha yes. This is branch on branch nonsense, that when cleaned up folded into this. I'll fix. Nice catch!